File: 1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        # sort nums so the addition of the first element and the last is minimized
        nums.sort()
        # track max sum
        max_sum = 0
        l, r = 0, len(nums) - 1
        while l < r:
            max_sum = max(max_sum, nums[l] + nums[r])
            l += 1
            r -= 1
        return max_sum

File: 1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.cpp
class Solution {
public:
    int minPairSum(vector<int>& nums) {
        std::sort(nums.begin(), nums.end());
        int l = 0;
        int r = nums.size() - 1;
        int max_sum = INT_MIN;
        while (l < r) {
            int sum = nums[l] + nums[r];
            max_sum = std::max(max_sum, sum);
            l += 1;
            r -= 1;
        }
        return max_sum;
    }
};